feat/gene/gene.go

```diff
@@ -218,52 +218,85 @@ func (t *CodingTranscript) Orientation() feat.Orientation {
 	return t.Orient
 }
 
-// UTR5start returns the start of the 5'UTR relative to the transcript.
-func (t *CodingTranscript) UTR5start() int {
+// UTR5 returns a feat.Feature that corresponds to the 5'UTR of the
+// transcript.
+func (t *CodingTranscript) UTR5() feat.Feature {
+	var start, end int
 	ori, _ := feat.BaseOrientationOf(t)
 	switch ori {
 	case feat.Forward:
-		return 0
+		start = 0
+		end = t.CDSstart
 	case feat.Reverse:
-		return t.CDSend
+		start = t.CDSend
+		end = t.Len()
+	default:
+		panic("gene: zero orientation for transcript")
+	}
+	return &TranscriptFeature{
+		Transcript: t,
+		Offset:     start,
+		Length:     end - start,
+		Orient:     feat.Forward,
 	}
-	panic("gene: zero orientation for transcript")
 }
 
-// UTR5end returns the end of the 5'UTR relative to the transcript.
-func (t *CodingTranscript) UTR5end() int {
+// CDS returns a feat.Feature that corresponds to the coding region of the
+// transcript.
+func (t *CodingTranscript) CDS() feat.Feature {
+	return &TranscriptFeature{
+		Transcript: t,
+		Offset:     t.CDSstart,
+		Length:     t.CDSend - t.CDSstart,
+		Orient:     feat.Forward,
+	}
+}
+
+// UTR3 returns a feat.Feature that corresponds to the 3'UTR of the
+// transcript.
+func (t *CodingTranscript) UTR3() feat.Feature {
+	var start, end int
 	ori, _ := feat.BaseOrientationOf(t)
 	switch ori {
 	case feat.Forward:
-		return t.CDSstart
+		start = t.CDSend
+		end = t.Len()
 	case feat.Reverse:
-		return t.Len()
+		start = 0
+		end = t.CDSstart
+	default:
+		panic("gene: zero orientation for transcript")
+	}
+	return &TranscriptFeature{
+		Transcript: t,
+		Offset:     start,
+		Length:     end - start,
+		Orient:     feat.Forward,
 	}
-	panic("gene: zero orientation for transcript")
+}
+
+// UTR5start returns the start of the 5'UTR relative to the transcript.
+// Deprecated: New code should use t.UTR5().Start() instead.
```

Comment:
Deprecation indicates that these methods will be removed at a later date. This won't happen, so please rewrite this as something like "UTR5start is shorthand for UTR5().Start()." Same for the methods below.